### ISSUE TITLE
Question about the permission of pika-bind script

### ISSUE THREAD

**Issue by topazus** [pika-org/pika#911]
https://github.com/pika-org/pika/blob/dc24ede8eb987a2c877874a13aa6ca809107cd31/CMakeLists.txt#L1759-L1764

With setting of `PERMISSIONS OWNER_READ GROUP_READ WORLD_READ OWNER_EXECUTE GROUP_EXECUTE WORLD_EXECUTE` to the file,  and I do the RPM packaging of the project. The results show that the file is write-protected regular file, when I want to remove the file from the RPM package.

**Comment by msimberg** [comment]
Hi @topazus, nice to hear that you're creating an RPM package! Are you saying `make install` (or the equivalent with `ninja`/`cmake`) is failing? What exactly is the command that is failing and with what output? Do you have a suggestion for how we can change the installation to work in your case?

**Comment by topazus** [comment]
> Are you saying `make install` (or the equivalent with `ninja`/`cmake`) is failing?

Previously, I want to delete the pika-bind file when make RPM packaging. Since I noticed that it is a helpful script, so I think it could be included in the package. But after installing the successful built package, its file permission `r-xr-xr-x` may be not correct compared with other similar executable files `rwxr-xr-x`.

```bash
❯ ll /usr/bin/gcc
-rwxr-xr-x. 3 root root 1043656 Dec  5 08:00 /usr/bin/gcc

❯ ll /usr/bin/pika-bind 
-r-xr-xr-x. 1 root root 7104 Jan  3 08:00 /usr/bin/pika-bind
```


> Do you have a suggestion for how we can change the installation to work in your case?

Deteling the two lines will result in the correct file permmission

https://github.com/pika-org/pika/blob/dc24ede8eb987a2c877874a13aa6ca809107cd31/CMakeLists.txt#L1762-L1763

**Comment by msimberg** [comment]
> Previously, I want to delete the pika-bind file when make RPM packaging. Since I noticed that it is a helpful script, so I think it could be included in the package.

Yeah, it would definitely be nice to include it. It's not always needed, but when it is it's nice to have around.

> But after installing the successful built package, its file permission `r-xr-xr-x` may be not correct compared with other similar executable files `rwxr-xr-x`.
> 
> ```shell
> ❯ ll /usr/bin/gcc
> -rwxr-xr-x. 3 root root 1043656 Dec  5 08:00 /usr/bin/gcc
> 
> ❯ ll /usr/bin/pika-bind 
> -r-xr-xr-x. 1 root root 7104 Jan  3 08:00 /usr/bin/pika-bind
> ```
> 
> > Do you have a suggestion for how we can change the installation to work in your case?
> 
> Deteling the two lines will result in the correct file permmission
> 
> https://github.com/pika-org/pika/blob/dc24ede8eb987a2c877874a13aa6ca809107cd31/CMakeLists.txt#L1762-L1763

I see, thanks for the explanation! I think this is the part of the documentation that we've missed: https://cmake.org/cmake/help/latest/command/install.html#programs. Specifically:

> The PROGRAMS form is identical to the FILES form except that the default permissions for the installed file also include OWNER_EXECUTE, GROUP_EXECUTE, and WORLD_EXECUTE. This form is intended to install programs that are not targets, such as shell scripts. Use the TARGETS form to install targets built within the project.

And thus leaving out the explicit permissions will lead to sane defaults. This post seems to have some explanation as to why it should be writable by root ("it's arbitrary"): https://unix.stackexchange.com/questions/340295/why-are-executables-in-e-g-usr-sbin-writable-by-root.

**Comment by msimberg** [comment]
@topazus I've just opened #915 to remove the explicit permissions. Would you mind having a look to see if it does what you'd expect? If merged, it should be available in the 0.22.0 release planned for early February.

**Comment by topazus** [comment]
I have tested the PR as a patch to build RPM package, it did as expected.
